compiler/rustc_lint/src/builtin.rs

```diff
@@ -797,21 +758,18 @@ impl<'tcx> LateLintPass<'tcx> for MissingDebugImplementations {
 
         let Some(debug) = cx.tcx.get_diagnostic_item(sym::Debug) else { return };
```

Comment:
Nit: this is usually true so it can be moved to after the `level == Level::Allow` condition.